### PR TITLE
hotfix: dataloaders - add unimplemented methods

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

there were a few critical bigs - unimplemented methods...

Fixes #1959

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes [if needed]?
- [x] Did you write any new necessary tests [no need for typos, docs]? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone are aligned!**
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by s-rog — APPROVED** [review]

File: pytorch_lightning/trainer/supporters.py [hunk 1]
```diff
@@ -279,14 +294,14 @@ def _calc_num_data(datasets: Union[Sequence, Mapping], mode: str) -> Union[int,
             length: the length of `CombinedDataset`
 
         """
-        if mode not in ['min_size', 'max_size_cycle']:
+        if mode not in CombinedDataset.COMPUTE_FUNCS.keys():
             raise ValueError(f"Invalid Mode: {mode}")
```
Comment (rohitgr7) [inline pytorch_lightning/trainer/supporters.py:299]: this is already checked in `__init__`?

File: pytorch_lightning/trainer/supporters.py [hunk 2]
```diff
@@ -251,19 +251,34 @@ class CombinedDataset(object):
...
             datasets: a sequence/mapping datasets. Can be a collections of torch.utils.Dataset,
                 Iterable or even None.
-            mode: whether to use the minimum number of batches in all samples or the maximum 
+            mode: whether to use the minimum number of batches in all samples or the maximum
                 number of batches in all samples.
 
         """
         self.datasets = datasets
+        if mode not in self.COMPUTE_FUNCS.keys():
+            raise ValueError(
```
Comment (rohitgr7) [inline pytorch_lightning/trainer/supporters.py:268]: raise `MisconfigurationException`?

**Review by tchaton — APPROVED** [review]

File: pytorch_lightning/trainer/supporters.py (hunk 1 above)
Comment (Borda) [reply pytorch_lightning/trainer/supporters.py:299]: yes, but here it is in the static method...

File: pytorch_lightning/trainer/supporters.py (hunk 2 above)
Comment (Borda) [reply pytorch_lightning/trainer/supporters.py]: done

**Review by ananyahjha93 — APPROVED** [review]
